### PR TITLE
FlareSystemsFirework: update to 1.0.1 solution

### PR DESCRIPTION
**Pull request by jctaillandier**
Fixes #

## Proposed Changes

  - Added Playbook
  - Changed wizard and instructions in Data Connector as well as Solution deployment (Zip)
  - The changes to the zip are already published in microsoft marketplace listing

### CODE REVIEW

**Comment by lior-tamir** [comment]
Hi,
Seems like the playbook includes only the Json definition of the playbook, while it should include a whole ARM template of:
* Workflow resource
* Resource for each API connection
* Parameters (such playbook name)

Also, I see there is a 3 levels For Each - Is it required? As it may be confusing/costly in case of multiple items.

**Comment by jctaillandier** [comment]
> * Workflow resource
> * Resource for each API connection
> * Parameters (such playbook name)

Could you give an example? I see [Fortinet playbooks](https://github.com/Azure/Azure-Sentinel/tree/f82f321872b2472be3a44041ef718f5b199922b7/Solutions/Fortinet-FortiGate/Playbooks) as well as [RiskIQ's](https://github.com/Azure/Azure-Sentinel/tree/f82f321872b2472be3a44041ef718f5b199922b7/Solutions/RiskIQ/Playbooks) only have a JSON and an .md

**Comment by lior-tamir** [comment]
In the azuredeploy.json files (of Fortinet/RiskIQ) there are more then just the logic app definition. Your Json includes only the "definition" field of the "properties" of the resource of "type": "Microsoft.Logic/workflows".

A "clean" template can be found here:
https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json

**Review by v-rucdu — CHANGES_REQUESTED** [review]

File: Solutions/FlareSystemsFirework/Playbooks/readme.md
Comment (jctaillandier) [reply Solutions/FlareSystemsFirework/Playbooks/readme.md:4]: Thanks!

**Comment by jctaillandier** [comment]
Struggling a bit to properly fill that azuredeploy template, you guys have any doc ?

**Comment by v-jayakal** [comment]
@jctaillandier - you can refer the template here https://github.com/Azure/Azure-Sentinel/tree/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template. please go thru readme.
You can also refer the other solutions with functions.

**Comment by v-jayakal** [comment]
@v-rucdu , can you please provide details on the validation failures?

**Comment by v-jayakal** [comment]
@lior-tamir - Can you please review the playbook? Data connector change is good. Once you approve, we will merge the PR, thanks.

**Review by v-jayakal — APPROVED** [review]

File: Solutions/FlareSystemsFirework/Data Connectors/Connector_REST_API_FlareSystemsFirework.json
Comment (v-rucdu) [inline Solutions/FlareSystemsFirework/Data Connectors/Connector_REST_API_FlareSystemsFirework.json:76]: Validation is failing because the Azure Configuration documentation link contains locale **en-us**. Please remove it.
suggestion for url that can be used is : **https://docs.microsoft.com/en-us/azure/sentinel/connect-data-sources**

**Comment by v-rucdu** [comment]
@lior-tamir : Could you please review the playbooks?

**Comment by lior-tamir** [comment]
Hi, 
Some notes:
1. Default value for playbook name parameter should not contain spaces (Since it is going to be a resource name in Azure, which cannot contain spaces)
2. Playbook deployment fails for me. I think it is because one of the Office 365 connection is missing from the "connections" sections and the resources section.
Error is:
    {
      "code": "InvalidTemplate",
      "message": "The template validation failed: 'The workflow definition parameter '$connections' at line '1' and column '176' is not valid: the 'type' property is not specified.'."
    }

Thanks!

**Comment by preetikr** [comment]
@jctaillandier - Please incorporate the new round of review feedback so that we can move this forward. Thanks

**Comment by v-rucdu** [comment]
@jctaillandier - Please resolve the review feedback.... Thanks!!!

**Comment by jctaillandier** [comment]
@v-rucdu @preetikr Sorry for the delay, should be fine now

File: Solutions/FlareSystemsFirework/Playbooks/azuredeploy.json
Comment (NikTripathi) [inline Solutions/FlareSystemsFirework/Playbooks/azuredeploy.json:12]: Name should not have spaces in between. You ca use '-' | '_' | '.' 
Please update it accordingly.

Comment (jctaillandier) [reply Solutions/FlareSystemsFirework/Playbooks/azuredeploy.json]: Sorry about that!

**Review by NikTripathi — CHANGES_REQUESTED** [review]
While deploying the playbook getting this error.
image

**Comment by jctaillandier** [comment]
@NikTripathi Should work now.. Otherwise not sure whats up!

**Comment by NikTripathi** [comment]
@lior-tamir : Could you please review the playbooks?

**Comment by lior-tamir** [comment]
Hi,
Did you manage to deploy the raw code of the playbooks?
I get an error:
{
  "code": "DeploymentFailed",
  "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
  "details": [
    {
      "code": "InvalidTemplate",
      "message": "The template validation failed: 'The workflow definition parameter '$connections' at line '1' and column '176' is not valid: the 'type' property is not specified.'."
    }
  ]
}

Also, noticed some errors in Office 365 connection:
1. Missing a resource for the Office 365 connection (same as Sentinel has a resource definition, and workflow depends on it)
2. The connection reference to Office 365 includes an unsupported authentication type: lines 274-278 are not supported by Outlook connector and should be removed.
3. This seems wrong syntax: "<...>" in 271,273 - probably should be same as in Sentinel connection.

**Comment by jctaillandier** [comment]
I cannot test it locally since
https://portal.azure.com/?feature.BringYourOwnConnector=true 
doesnt show the import option anymore

**Comment by v-jayakal** [comment]
@jctaillandier - I could import json file using https://portal.azure.com/?feature.BringYourOwnConnector=true

image
Can you please try once again?

**Comment by jctaillandier** [comment]
Yes it works for data connector, I was talking about playbook import. Isn't that the problematic piece ?

**Comment by jctaillandier** [comment]
Right, data connector doesnt seem to be the issue, I was talking about playbook template import

**Comment by v-jayakal** [comment]
@lior-tamir - Can you please help here on importing playbooks, thanks.

**Comment by anki-narravula** [comment]
Hi @jctaillandier - there are few issues with azuredeploy.json template. I have corrected and attached the file here, can you please submit the modified file. Also regarding the feature flag not working for you, what is the region of your resource group / sentinel is in ? So that I will test for that 
[FlareFirework_azuredeploy.zip](https://github.com/Azure/Azure-Sentinel/files/7359631/FlareFirework_azuredeploy.zip)

**Comment by jctaillandier** [comment]
Subscription id: 7bd8ef24-f62e-4ed8-ba05-962f2b23bb4f
Workspace: 0da0c3cf-7422-40bf-8c40-0252fdb8c30b
Cheers!

**Comment by v-maudan** [comment]
@jctaillandier, To pass PR validations, Can you please merge your branch with master. Thanks!

**Comment by jctaillandier** [comment]
You mean rebase on master ? I cannot do a merge request since missing one more reviewer!

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**
